compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -880,6 +880,97 @@ impl<'a> Parser<'a> {
         }
     }
 
+    pub(super) fn check_anonymous_enum(&mut self, span: Span, tys: &[P<Ty>]) -> PResult<'a, ()> {
+        use std::fmt::Write;
+        if tys.len() <= 1 {
+            return Ok(());
+        }
+
+        fn variant_name_and_ty<'a>(
+            ty: &'a Ty,
+            lifetimes: &mut FxHashSet<&'a str>,
+        ) -> Option<(String, String)> {
+            match &ty.kind {
+                TyKind::Path(_, path) => {
+                    let mut name = String::new();
+                    let mut ty_string = String::new();
+
+                    if let Some(seg) = path.segments.iter().last() {
+                        name.push_str(seg.ident.name.as_str());
+                        ty_string.push_str(seg.ident.name.as_str());
+
+                        if let Some(_args) = &seg.args {
+                            ty_string.push('<');
+                            ty_string.push_str("...");
```

Comment:
Why don't we print the whole type?

Reply:
it's more complicated, for example `Vec<impl Send> | Vec<(i32, i64)>`. We can not make a enum variant name for these type.

Reply:
As you propose, we can use `Variant{N}` as the variant name